user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -139,12 +151,125 @@ protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser
             scramAcls = user.getSimpleAclRules();
         }
 
+        KafkaUserStatus userStatus = new KafkaUserStatus();
         CompositeFuture.join(
                 scramShaCredentialOperator.reconcile(user.getName(), password),
-                secretOperations.reconcile(namespace, user.getSecretName(), desired),
+                reconcileSecretAndSetStatus(namespace, user, desired, userStatus),
                 aclOperations.reconcile(KafkaUserModel.getTlsUserName(userName), tlsAcls),
                 aclOperations.reconcile(KafkaUserModel.getScramUserName(userName), scramAcls))
-                .map((Void) null).setHandler(handler);
+                .compose(reconciliationResult -> {
+                    Condition readyCondition;
+                    if (kafkaUser.getMetadata().getGeneration() != null)    {
+                        userStatus.setObservedGeneration(kafkaUser.getMetadata().getGeneration());
+                    }
+
+                    if (reconciliationResult.succeeded()) {
+                        readyCondition = new ConditionBuilder()
+                                .withNewLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                .withNewType("Ready")
+                                .withNewStatus("True")
+                                .build();
+                    } else {
+                        readyCondition = new ConditionBuilder()
+                                .withNewLastTransitionTime(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(dateSupplier()))
+                                .withNewType("NotReady")
+                                .withNewStatus("True")
+                                .withNewReason(reconciliationResult.cause().getClass().getSimpleName())
+                                .withNewMessage(reconciliationResult.cause().getMessage())
+                                .build();
+                    }
```

Comment:
This is the same code as we have in KAO. Can we have a common method on some util class in operator-common for building `Ready`/`NotReady` `Conditions` from a `Future`?